common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -16,57 +16,61 @@ type StaticType = { a: string; b: string };
+const indexedRecordOfStrings: IndexSignatureType = someObjWithStaticType;
+indexedRecordOfStrings.a; // ok: Accessing index property 'a' without requiring a particular result
+indexedRecordOfStrings.a.length; // defect: Accessing length of index property 'a', but 'a' might not be present
+indexedRecordOfStrings["a"]; // ok: Accessing property 'a' using bracket notation
+indexedRecordOfStrings["a"].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[a].length; // defect: Accessing length of index property 'a' using bracket notation, but 'a' might not be present
+indexedRecordOfStrings[b].length; // defect: Accessing length of index property 'b' using bracket notation, but 'b' might not be present
+indexedRecordOfStrings[c].length; // defect: Accessing length of missing property 'c' using bracket notation
+indexedRecordOfStrings.a?.length; // ok: Using optional chaining to access length safely handles 'undefined'
+indexedRecordOfStrings.a!.length; // ok: Using non null assert to access length safely handles 'undefined'
+indexedRecordOfStrings["a"]?.length; // ok: Using optional chaining to access length using bracket notation safely handles 'undefined'
+indexedRecordOfStrings["a"]!.length; // ok: Using non null assert to access length using bracket notation safely handles 'undefined'
 
-if (someObjWithPotenciallyUndefinedProperties.a) {
-	someObjWithPotenciallyUndefinedProperties.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+if (indexedRecordOfStrings.a) {
+	indexedRecordOfStrings.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
 }
 
-for (const key in someObjWithPotenciallyUndefinedProperties) {
-	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
-	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+for (const key in indexedRecordOfStrings) {
+	indexedRecordOfStrings[key]; // ok: Accessing property while looping though records which acts like a `has` property check
+	indexedRecordOfStrings[key].length; // ok: Accessing property while looping though records which acts like a `has` property check
 }
 
-for (const [key, value] of Object.entries(someObjWithPotenciallyUndefinedProperties)) {
-	value; // ok: Accessing value from entries
-	value.length; // ok: Accessing length of value from entries, entries should be string values
-	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
-	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+for (const [key, value] of Object.entries(indexedRecordOfStrings)) {
+	value; // ok: Access is always okay because the result is not checked
+	value.length; // ok: Accessing property while looping though records which acts like a `has` property check
```

Comment:
This is not correct reasoning for okay on lines 59 and 60. There is no indexed access here. This is simple use of value that was already provided by `Object.entries`.
Remove line 59 as not useful.
For line 60, update to say // ok: Object.entries provides only present values